wsmaster/che-core-api-user/src/main/java/org/eclipse/che/api/user/server/jpa/JpaUserDao.java

```diff
@@ -160,6 +164,32 @@ public UserImpl getByEmail(String email) throws NotFoundException, ServerExcepti
         }
     }
 
+    @Override
+    @Transactional
+    public Page<UserImpl> getAll(int maxItems, int skipCount) throws ServerException {
+        checkArgument(maxItems >= 0, "The number of items to return can't be negative.");
+        checkArgument(skipCount >= 0, "The number of items to skip can't be negative.");
+        try {
+            List<UserImpl> list = managerProvider.get()
+                                                 .createNamedQuery("User.getAll", UserImpl.class)
+                                                 .setMaxResults(maxItems)
+                                                 .setFirstResult(skipCount)
+                                                 .getResultList();
+            return new Page<>(list, skipCount, maxItems, getTotalCount());
+        } catch (RuntimeException x) {
+            throw new ServerException(x.getLocalizedMessage(), x);
+        }
+    }
+
+    @Transactional
+    protected int getTotalCount() throws ServerException {
+        try {
+            return ((Number) managerProvider.get().createNamedQuery("User.getTotalCount").getSingleResult()).intValue();
```

Comment:
use Long.class as second argument in `.createNamedQuery` method, and please remove the cast